<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Make a dataset for a condition of interest. — makeSEforCondition • curatedMetagenomicAnalyses</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Make a dataset for a condition of interest. — makeSEforCondition"><meta name="description" content="Make a dataset for a condition of interest."><meta property="og:description" content="Make a dataset for a condition of interest."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">curatedMetagenomicAnalyses</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html">Home</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-analyses" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Analyses</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-analyses"><li><h6 class="dropdown-header" data-toc-skip>R Vignettes</h6></li>
    <li><a class="dropdown-item" href="../articles/MLdatasets.html">Create datasets for machine learning</a></li>
    <li><a class="dropdown-item" href="../articles/explorecirrhosis.html">Exploration of the liver cirrhosis dataset</a></li>
    <li><a class="dropdown-item" href="../articles/identify_CRC_species.html">Colorectal cancer stool microbiome species prevalence (dynamic table)</a></li>
    <li><a class="dropdown-item" href="../articles/Age_metaanalysis_vignette.html">Meta-analysis of age-related microbial species using cMD3</a></li>
    <li><a class="dropdown-item" href="../articles/Sex_metaanalysis_vignette.html">Meta-analysis of sex-related microbial species using cMD3</a></li>
    <li><a class="dropdown-item" href="../articles/NUI-Galway-Metagenomics-Workshop.html">NUI Galway Metagenomics Workshop</a></li>
    <li><h6 class="dropdown-header" data-toc-skip>Python Notebooks</h6></li>
    <li><a class="dropdown-item" href="https://github.com/waldronlab/curatedMetagenomicAnalyses/blob/main/vignettes/sexContrastMicrobiomeAnalysis.ipynb">Sex-related differences in the human microbiome using cMD3 and Python3</a></li>
    <li><h6 class="dropdown-header" data-toc-skip>Supplementary Materials</h6></li>
    <li><a class="dropdown-item" href="https://github.com/waldronlab/curatedMetagenomicAnalyses/blob/main/vignettes/installation.ipynb">Installing Python dependencies in Linux (Python Notebook)</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="nav-link" href="https://github.com/waldronlab/curatedMetagenomicAnalyses"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Make a dataset for a condition of interest.</h1>
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/curatedMetagenomicAnalyses/blob/main/R/makeSEforCondition.R"><code>R/makeSEforCondition.R</code></a></small>
      <div class="d-none name"><code>makeSEforCondition.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Make a dataset for a condition of interest.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">makeSEforCondition</span><span class="op">(</span></span>
<span>  <span class="va">condition</span>,</span>
<span>  removestudies <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  dataType <span class="op">=</span> <span class="st">"relative_abundance"</span>,</span>
<span>  counts <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-condition">condition<a class="anchor" aria-label="anchor" href="#arg-condition"></a></dt>
<dd><p>Condition of study for which to build a case-control dataset.
See "study_condition" column of the <code>sampleMetadata</code> object.</p></dd>


<dt id="arg-removestudies">removestudies<a class="anchor" aria-label="anchor" href="#arg-removestudies"></a></dt>
<dd><p>Any studies not to be included (default: NULL)</p></dd>


<dt id="arg-datatype">dataType<a class="anchor" aria-label="anchor" href="#arg-datatype"></a></dt>
<dd><p>Type of metagenomic data to return, see <code>?curatedMetagenomicData</code></p></dd>


<dt id="arg-counts">counts<a class="anchor" aria-label="anchor" href="#arg-counts"></a></dt>
<dd><p>Convert to something resembling counts, by multiplying through
by read depth?</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>a (Tree)SummarizedExperiment containing merged data from 1+ studies</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function finds datasets that contain the condition of interest, returns
those datasets, and filters them to contain only samples of the condition or
controls. These datasets are then merged into a single
(Tree)SummarizedExperiment. Controls from other datasets are not included.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">makeSEforCondition</span><span class="op">(</span><span class="st">"STH"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> $`2021-10-14.RosaBA_2018.relative_abundance`</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> dropping rows without rowTree matches:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   k__Bacteria|p__Actinobacteria|c__Coriobacteriia|o__Coriobacteriales|f__Atopobiaceae|g__Olsenella|s__Olsenella_profusa</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   k__Bacteria|p__Actinobacteria|c__Coriobacteriia|o__Coriobacteriales|f__Coriobacteriaceae|g__Collinsella|s__Collinsella_stercoris</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   k__Bacteria|p__Firmicutes|c__Bacilli|o__Bacillales|f__Bacillales_unclassified|g__Gemella|s__Gemella_bergeri</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Carnobacteriaceae|g__Granulicatella|s__Granulicatella_elegans</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   k__Bacteria|p__Firmicutes|c__Erysipelotrichia|o__Erysipelotrichales|f__Erysipelotrichaceae|g__Bulleidia|s__Bulleidia_extructa</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae|g__Sutterella|s__Sutterella_parvirubra</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> $`2021-10-14.RubelMA_2020.relative_abundance`</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> dropping rows without rowTree matches:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   k__Bacteria|p__Actinobacteria|c__Coriobacteriia|o__Coriobacteriales|f__Coriobacteriaceae|g__Collinsella|s__Collinsella_stercoris</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   k__Bacteria|p__Firmicutes|c__Clostridia|o__Clostridiales|f__Ruminococcaceae|g__Ruminococcus|s__Ruminococcus_champanellensis</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae|g__Sutterella|s__Sutterella_parvirubra</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 496 199 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): relative_abundance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(496):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   k__Bacteria|p__Actinobacteria|c__Actinobacteria|o__Bifidobacteriales|f__Bifidobacteriaceae|g__Bifidobacterium|s__Bifidobacterium_adolescentis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   k__Bacteria|p__Firmicutes|c__Clostridia|o__Clostridiales|f__Lachnospiraceae|g__Roseburia|s__Roseburia_faecis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   k__Bacteria|p__Bacteroidetes|c__Bacteroidia|o__Bacteroidales|f__Porphyromonadaceae|g__Porphyromonas|s__Porphyromonas_gingivalis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   k__Bacteria|p__Proteobacteria|c__Gammaproteobacteria|o__Enterobacterales|f__Morganellaceae|g__Morganella|s__Morganella_morganii</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(7): superkingdom phylum ... genus species</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(199): U_VS-3059-508 U_VS-1592-367 ... CM.94_WGS CM.97_WGS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(27): study_name subject_id ... lifestyle</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   uncurated_metadata</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: a LinkDataFrame (496 rows)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: 1 phylo tree(s) (10430 leaves)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Levi Waldron, Paolo Manghi, Davide Galzato, Lucas Schiffer, Jennifer Wokaty.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer></div>





  </body></html>

