<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Exploration of the liver cirrhosis dataset â€¢ curatedMetagenomicAnalyses</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Exploration of the liver cirrhosis dataset">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">curatedMetagenomicAnalyses</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html">Home</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-analyses" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Analyses</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-analyses">
<li><h6 class="dropdown-header" data-toc-skip>R Vignettes</h6></li>
    <li><a class="dropdown-item" href="../articles/MLdatasets.html">Create datasets for machine learning</a></li>
    <li><a class="dropdown-item" href="../articles/explorecirrhosis.html">Exploration of the liver cirrhosis dataset</a></li>
    <li><a class="dropdown-item" href="../articles/identify_CRC_species.html">Colorectal cancer stool microbiome species prevalence (dynamic table)</a></li>
    <li><a class="dropdown-item" href="../articles/Age_metaanalysis_vignette.html">Meta-analysis of age-related microbial species using cMD3</a></li>
    <li><a class="dropdown-item" href="../articles/Sex_metaanalysis_vignette.html">Meta-analysis of sex-related microbial species using cMD3</a></li>
    <li><a class="dropdown-item" href="../articles/NUI-Galway-Metagenomics-Workshop.html">NUI Galway Metagenomics Workshop</a></li>
    <li><h6 class="dropdown-header" data-toc-skip>Python Notebooks</h6></li>
    <li><a class="dropdown-item" href="https://github.com/waldronlab/curatedMetagenomicAnalyses/blob/main/vignettes/sexContrastMicrobiomeAnalysis.ipynb">Sex-related differences in the human microbiome using cMD3 and Python3</a></li>
    <li><h6 class="dropdown-header" data-toc-skip>Supplementary Materials</h6></li>
    <li><a class="dropdown-item" href="https://github.com/waldronlab/curatedMetagenomicAnalyses/blob/main/vignettes/installation.ipynb">Installing Python dependencies in Linux (Python Notebook)</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="nav-link" href="https://github.com/waldronlab/curatedMetagenomicAnalyses"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<link href="explorecirrhosis_files/table1-1.0/table1_defaults.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Exploration of the liver cirrhosis dataset</h1>
                        <h4 data-toc-skip class="author">Levi
Waldron</h4>
            
            <h4 data-toc-skip class="date">July 19, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/curatedMetagenomicAnalyses/blob/main/vignettes/explorecirrhosis.Rmd"><code>vignettes/explorecirrhosis.Rmd</code></a></small>
      <div class="d-none name"><code>explorecirrhosis.Rmd</code></div>
    </div>

    
    
<p>Packages used here:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/curatedMetagenomicData" class="external-link">curatedMetagenomicData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/curatedMetagenomicAnalyses">curatedMetagenomicAnalyses</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="cirrhosis-exploratory-analysis">Cirrhosis exploratory analysis<a class="anchor" aria-label="anchor" href="#cirrhosis-exploratory-analysis"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu">curatedMetagenomicAnalyses</span><span class="fu">::</span><span class="fu"><a href="../reference/makeSEforCondition.html">makeSEforCondition</a></span><span class="op">(</span><span class="st">"cirrhosis"</span>, removestudies <span class="op">=</span> <span class="st">"HMP_2019_ibdmdb"</span>, dataType <span class="op">=</span> <span class="st">"relative_abundance"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## $`2021-10-14.LoombaR_2017.relative_abundance`</span></span>
<span><span class="co">## dropping rows without rowTree matches:</span></span>
<span><span class="co">##   k__Bacteria|p__Actinobacteria|c__Coriobacteriia|o__Coriobacteriales|f__Coriobacteriaceae|g__Collinsella|s__Collinsella_stercoris</span></span>
<span><span class="co">##   k__Bacteria|p__Firmicutes|c__Bacilli|o__Bacillales|f__Bacillales_unclassified|g__Gemella|s__Gemella_bergeri</span></span>
<span><span class="co">##   k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Carnobacteriaceae|g__Granulicatella|s__Granulicatella_elegans</span></span>
<span><span class="co">##   k__Bacteria|p__Firmicutes|c__Erysipelotrichia|o__Erysipelotrichales|f__Erysipelotrichaceae|g__Bulleidia|s__Bulleidia_extructa</span></span>
<span><span class="co">##   k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae|g__Sutterella|s__Sutterella_parvirubra</span></span>
<span><span class="co">##   k__Bacteria|p__Synergistetes|c__Synergistia|o__Synergistales|f__Synergistaceae|g__Cloacibacillus|s__Cloacibacillus_evryensis</span></span></code></pre>
<pre><code><span><span class="co">## $`2021-03-31.QinN_2014.relative_abundance`</span></span>
<span><span class="co">## dropping rows without rowTree matches:</span></span>
<span><span class="co">##   k__Bacteria|p__Actinobacteria|c__Coriobacteriia|o__Coriobacteriales|f__Atopobiaceae|g__Olsenella|s__Olsenella_profusa</span></span>
<span><span class="co">##   k__Bacteria|p__Actinobacteria|c__Coriobacteriia|o__Coriobacteriales|f__Coriobacteriaceae|g__Collinsella|s__Collinsella_stercoris</span></span>
<span><span class="co">##   k__Bacteria|p__Actinobacteria|c__Coriobacteriia|o__Coriobacteriales|f__Coriobacteriaceae|g__Enorma|s__[Collinsella]_massiliensis</span></span>
<span><span class="co">##   k__Bacteria|p__Firmicutes|c__Bacilli|o__Bacillales|f__Bacillales_unclassified|g__Gemella|s__Gemella_bergeri</span></span>
<span><span class="co">##   k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Carnobacteriaceae|g__Granulicatella|s__Granulicatella_elegans</span></span>
<span><span class="co">##   k__Bacteria|p__Firmicutes|c__Clostridia|o__Clostridiales|f__Ruminococcaceae|g__Ruminococcus|s__Ruminococcus_champanellensis</span></span>
<span><span class="co">##   k__Bacteria|p__Firmicutes|c__Erysipelotrichia|o__Erysipelotrichales|f__Erysipelotrichaceae|g__Bulleidia|s__Bulleidia_extructa</span></span>
<span><span class="co">##   k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae|g__Sutterella|s__Sutterella_parvirubra</span></span>
<span><span class="co">##   k__Bacteria|p__Synergistetes|c__Synergistia|o__Synergistales|f__Synergistaceae|g__Cloacibacillus|s__Cloacibacillus_evryensis</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/benjaminrich/table1" class="external-link">table1</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Registered S3 method overwritten by 'table1':</span></span>
<span><span class="co">##   method    from</span></span>
<span><span class="co">##   [.indexed plyr</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'table1'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     units, units&lt;-</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu">table1</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/table1/man/label.html" class="external-link">label</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">study_name</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"Study Name"</span></span>
<span><span class="fu">table1</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/table1/man/label.html" class="external-link">label</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">body_site</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"Body site"</span></span>
<span><span class="fu">table1</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/table1/man/label.html" class="external-link">label</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">study_condition</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"Study condition"</span></span>
<span><span class="fu">table1</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/table1/man/label.html" class="external-link">label</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">antibiotics_current_use</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"Current Antibiotics Use"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/table1/man/units.html" class="external-link">units</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">age</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"Years"</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/table1/man/table1.html" class="external-link">table1</a></span><span class="op">(</span><span class="op">~</span> <span class="va">.</span> <span class="op">-</span>  <span class="va">subject_id</span> <span class="op">|</span> <span class="va">study_name</span>, data <span class="op">=</span> <span class="va">df</span><span class="op">)</span></span></code></pre></div>
<div class="Rtable1"><table class="table Rtable1">
<thead><tr>
<th class="rowlabel firstrow lastrow"></th>
<th class="firstrow lastrow"><span class="stratlabel">LoombaR_2017<br><span class="stratn">(N=45)</span></span></th>
<th class="firstrow lastrow"><span class="stratlabel">QinN_2014<br><span class="stratn">(N=237)</span></span></th>
<th class="firstrow lastrow"><span class="stratlabel">Overall<br><span class="stratn">(N=282)</span></span></th>
</tr></thead>
<tbody>
<tr>
<td class="rowlabel firstrow"><span class="varlabel">Body site</span></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel lastrow">stool</td>
<td class="lastrow">45 (100%)</td>
<td class="lastrow">237 (100%)</td>
<td class="lastrow">282 (100%)</td>
</tr>
<tr>
<td class="rowlabel firstrow"><span class="varlabel">Current Antibiotics Use</span></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">no</td>
<td>0 (0%)</td>
<td>237 (100%)</td>
<td>237 (84.0%)</td>
</tr>
<tr>
<td class="rowlabel lastrow">Missing</td>
<td class="lastrow">45 (100%)</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">45 (16.0%)</td>
</tr>
<tr>
<td class="rowlabel firstrow"><span class="varlabel">Study condition</span></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">cirrhosis</td>
<td>9 (20.0%)</td>
<td>123 (51.9%)</td>
<td>132 (46.8%)</td>
</tr>
<tr>
<td class="rowlabel lastrow">control</td>
<td class="lastrow">36 (80.0%)</td>
<td class="lastrow">114 (48.1%)</td>
<td class="lastrow">150 (53.2%)</td>
</tr>
<tr>
<td class="rowlabel firstrow"><span class="varlabel">disease</span></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">cirrhosis</td>
<td>9 (20.0%)</td>
<td>9 (3.8%)</td>
<td>18 (6.4%)</td>
</tr>
<tr>
<td class="rowlabel">fatty_liver</td>
<td>36 (80.0%)</td>
<td>0 (0%)</td>
<td>36 (12.8%)</td>
</tr>
<tr>
<td class="rowlabel">ascites;cirrhosis</td>
<td>0 (0%)</td>
<td>9 (3.8%)</td>
<td>9 (3.2%)</td>
</tr>
<tr>
<td class="rowlabel">ascites;cirrhosis;hepatitis</td>
<td>0 (0%)</td>
<td>55 (23.2%)</td>
<td>55 (19.5%)</td>
</tr>
<tr>
<td class="rowlabel">ascites;cirrhosis;hepatitis;schistosoma</td>
<td>0 (0%)</td>
<td>3 (1.3%)</td>
<td>3 (1.1%)</td>
</tr>
<tr>
<td class="rowlabel">ascites;cirrhosis;hepatitis;wilson</td>
<td>0 (0%)</td>
<td>1 (0.4%)</td>
<td>1 (0.4%)</td>
</tr>
<tr>
<td class="rowlabel">ascites;cirrhosis;schistosoma</td>
<td>0 (0%)</td>
<td>1 (0.4%)</td>
<td>1 (0.4%)</td>
</tr>
<tr>
<td class="rowlabel">ascites;cirrhosis;wilson</td>
<td>0 (0%)</td>
<td>1 (0.4%)</td>
<td>1 (0.4%)</td>
</tr>
<tr>
<td class="rowlabel">cirrhosis;hepatitis</td>
<td>0 (0%)</td>
<td>43 (18.1%)</td>
<td>43 (15.2%)</td>
</tr>
<tr>
<td class="rowlabel">healthy</td>
<td>0 (0%)</td>
<td>114 (48.1%)</td>
<td>114 (40.4%)</td>
</tr>
<tr>
<td class="rowlabel lastrow">hepatitis</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">1 (0.4%)</td>
<td class="lastrow">1 (0.4%)</td>
</tr>
<tr>
<td class="rowlabel firstrow"><span class="varlabel">age<span class="varunits"> (Years)</span></span></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">Mean (SD)</td>
<td>NA (NA)</td>
<td>46.5 (10.9)</td>
<td>46.5 (10.9)</td>
</tr>
<tr>
<td class="rowlabel">Median [Min, Max]</td>
<td>NA [NA, NA]</td>
<td>45.0 [18.0, 78.0]</td>
<td>45.0 [18.0, 78.0]</td>
</tr>
<tr>
<td class="rowlabel lastrow">Missing</td>
<td class="lastrow">45 (100%)</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">45 (16.0%)</td>
</tr>
<tr>
<td class="rowlabel firstrow"><span class="varlabel">age_category</span></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">adult</td>
<td>36 (80.0%)</td>
<td>221 (93.2%)</td>
<td>257 (91.1%)</td>
</tr>
<tr>
<td class="rowlabel">senior</td>
<td>9 (20.0%)</td>
<td>15 (6.3%)</td>
<td>24 (8.5%)</td>
</tr>
<tr>
<td class="rowlabel lastrow">schoolage</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">1 (0.4%)</td>
<td class="lastrow">1 (0.4%)</td>
</tr>
<tr>
<td class="rowlabel firstrow"><span class="varlabel">gender</span></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">female</td>
<td>0 (0%)</td>
<td>81 (34.2%)</td>
<td>81 (28.7%)</td>
</tr>
<tr>
<td class="rowlabel">male</td>
<td>0 (0%)</td>
<td>156 (65.8%)</td>
<td>156 (55.3%)</td>
</tr>
<tr>
<td class="rowlabel lastrow">Missing</td>
<td class="lastrow">45 (100%)</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">45 (16.0%)</td>
</tr>
<tr>
<td class="rowlabel firstrow"><span class="varlabel">country</span></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">USA</td>
<td>45 (100%)</td>
<td>0 (0%)</td>
<td>45 (16.0%)</td>
</tr>
<tr>
<td class="rowlabel lastrow">CHN</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">237 (100%)</td>
<td class="lastrow">237 (84.0%)</td>
</tr>
</tbody>
</table></div>
<p>There seems to be strong correlation between study condition and
current antibiotics use, raising questions of causality direction in
these predictions:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">study_condition</span>, <span class="va">df</span><span class="op">$</span><span class="va">antibiotics_current_use</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##            </span></span>
<span><span class="co">##              no</span></span>
<span><span class="co">##   cirrhosis 123</span></span>
<span><span class="co">##   control   114</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="compositionality">Compositionality<a class="anchor" aria-label="anchor" href="#compositionality"></a>
</h2>
<p>Many, but not all, columns add up to 100%:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">##   97.04   99.98  100.00   99.92  100.00  100.00</span></span></code></pre>
<p>The observation with minimum sum of relative abundances is:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.min.html" class="external-link">which.min</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 1 row and 31 columns</span></span>
<span><span class="co">##       study_name  subject_id   body_site antibiotics_current_use</span></span>
<span><span class="co">##      &lt;character&gt; &lt;character&gt; &lt;character&gt;             &lt;character&gt;</span></span>
<span><span class="co">## LV-1   QinN_2014        LV-1       stool                      no</span></span>
<span><span class="co">##      study_condition             disease       age age_category      gender</span></span>
<span><span class="co">##          &lt;character&gt;         &lt;character&gt; &lt;integer&gt;  &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">## LV-1       cirrhosis cirrhosis;hepatitis        42        adult        male</span></span>
<span><span class="co">##          country non_westernized sequencing_platform DNA_extraction_kit</span></span>
<span><span class="co">##      &lt;character&gt;     &lt;character&gt;         &lt;character&gt;        &lt;character&gt;</span></span>
<span><span class="co">## LV-1         CHN              no       IlluminaHiSeq                 NA</span></span>
<span><span class="co">##             PMID number_reads number_bases minimum_read_length</span></span>
<span><span class="co">##      &lt;character&gt;    &lt;integer&gt;    &lt;numeric&gt;           &lt;integer&gt;</span></span>
<span><span class="co">## LV-1    25079328     34522116   3452211600                 100</span></span>
<span><span class="co">##      median_read_length      NCBI_accession      curator       BMI</span></span>
<span><span class="co">##               &lt;integer&gt;         &lt;character&gt;  &lt;character&gt; &lt;numeric&gt;</span></span>
<span><span class="co">## LV-1                100 ERR528291;ERR528292 Paolo_Manghi     19.03</span></span>
<span><span class="co">##      disease_subtype antibiotics_family  creatine  albumine     alcohol</span></span>
<span><span class="co">##          &lt;character&gt;        &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;character&gt;</span></span>
<span><span class="co">## LV-1   HBV;cirrhosis                 NA        65      31.2         yes</span></span>
<span><span class="co">##      disease_stage bilirubin prothrombin_time       inr       ctp</span></span>
<span><span class="co">##        &lt;character&gt; &lt;numeric&gt;        &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;</span></span>
<span><span class="co">## LV-1            NA      1.82               NA      1.18         6</span></span></code></pre>
<p>Note, this a result of species where phylogenetic information was
unavailable in building of the <code>TreeSummarizedExperiment</code>. It
would be possible to take data directly from <code>ExperimentHub</code>
to avoid the data loss, but the data loss is rare and small and probably
unimportant for most purposes.</p>
</div>
<div class="section level2">
<h2 id="siamcat-analysis">SIAMCAT analysis<a class="anchor" aria-label="anchor" href="#siamcat-analysis"></a>
</h2>
<p>From the <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/SIAMCAT/inst/doc/SIAMCAT_vignette.html" class="external-link">SIAMCAT
vignette</a>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"SIAMCAT"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"SIAMCAT"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SIAMCAT</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">labs</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/SIAMCAT/man/create.label.html" class="external-link">create.label</a></span><span class="op">(</span>meta <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span>,</span>
<span>               label <span class="op">=</span> <span class="st">'study_condition'</span>,</span>
<span>               case <span class="op">=</span> <span class="st">'cirrhosis'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Label used as case:</span></span>
<span><span class="co">##    cirrhosis</span></span>
<span><span class="co">## Label used as control:</span></span>
<span><span class="co">##    control</span></span></code></pre>
<pre><code><span><span class="co">## + finished create.label.from.metadata in 0.022 s</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SIAMCAT/man/siamcat.html" class="external-link">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">/</span><span class="fl">100</span>,</span>
<span>    label<span class="op">=</span><span class="va">labs</span>,</span>
<span>    meta<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## + starting validate.data</span></span></code></pre>
<pre><code><span><span class="co">## +++ checking overlap between labels and features</span></span></code></pre>
<pre><code><span><span class="co">## + Keeping labels of 282 sample(s).</span></span></code></pre>
<pre><code><span><span class="co">## +++ checking sample number per class</span></span></code></pre>
<pre><code><span><span class="co">## +++ checking overlap between samples and metadata</span></span></code></pre>
<pre><code><span><span class="co">## + finished validate.data in 0.066 s</span></span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">sc.obj</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## siamcat-class object</span></span>
<span><span class="co">## label()                Label object:         150 control and 132 cirrhosis samples</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## contains phyloseq-class experiment-level object @phyloseq:</span></span>
<span><span class="co">## phyloseq@otu_table()   OTU Table:            [ 697 taxa and 282 samples ]</span></span>
<span><span class="co">## phyloseq@sam_data()    Sample Data:          [ 282 samples by 31 sample variables ]</span></span></code></pre>
<p>Unsupervised filtering:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SIAMCAT/man/filter.features.html" class="external-link">filter.features</a></span><span class="op">(</span><span class="va">sc.obj</span>,</span>
<span>    filter.method <span class="op">=</span> <span class="st">'abundance'</span>,</span>
<span>    cutoff <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Features successfully filtered</span></span></code></pre>
<div class="section level3">
<h3 id="association-testing">Association testing<a class="anchor" aria-label="anchor" href="#association-testing"></a>
</h3>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SIAMCAT/man/check.associations.html" class="external-link">check.associations</a></span><span class="op">(</span></span>
<span>    <span class="va">sc.obj</span>,</span>
<span>    alpha <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>    mult.corr <span class="op">=</span> <span class="st">"fdr"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SIAMCAT/man/association.plot.html" class="external-link">association.plot</a></span><span class="op">(</span><span class="va">sc.obj</span>,</span>
<span>    sort.by <span class="op">=</span> <span class="st">'fc'</span>,</span>
<span>    plot.type <span class="op">=</span> <span class="st">"quantile.box"</span>,</span>
<span>    panels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"fc"</span>, <span class="st">"prevalence"</span>, <span class="st">"auroc"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## ### ATTENTION: Not plotting to a pdf-file.</span></span>
<span><span class="co">## ### The plot is optimized for landscape DIN-A4 (or similar) layout.</span></span>
<span><span class="co">## ### Please make sure that your plotting region is large enough!!!</span></span>
<span><span class="co">## ### Use at your own risk...</span></span></code></pre>
<pre><code><span><span class="co">## Are you sure that you want to continue? (Yes/no/cancel)</span></span></code></pre>
<p><img src="explorecirrhosis_files/figure-html/unnamed-chunk-11-1.png" class="r-plt" width="700"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Levi Waldron, Paolo Manghi, Davide Galzato, Lucas Schiffer, Jennifer Wokaty.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
